lib/api/src/module.rs

```diff
@@ -217,6 +217,21 @@ impl Module {
         Ok(())
     }
 
+    /// See [`Self::dangerous_deserialize_unchecked`].
+    ///
+    /// # Safety
+    /// See [`Self::dangerous_deserialize_unchecked`].
+    #[deprecated(
+        since = "3.2.0",
+        note = "Use `deserialize_checked` or `dangerous_deserialize_unchecked` instead"
+    )]
```

Comment:
I don't think we need the dangerous here since it's already unsafe? We can just keep using the normal deserialize?

Reply:
Yeah, I also thought this was code in the VM. Let's not do a breaking change regarding deserialize

Reply:
I really feel very few users should use this function, unless they are really, really sure they know what they are doing - so it should also have a scary name.

Reply:
Let's put all breaking changes for 4.0